src/HERMIT/API/Dictionary/Reasoning.hs

```diff
@@ -173,29 +220,50 @@ extensionalityWith nm = Transform $ method "extensionalityWith" [toJSON nm]
 extensionality :: Rewrite LCore
 extensionality = Transform $ method "extensionality" []
 
--- | Apply a transformation to the LHS of a quantified clause.
-lhsT :: Transform LCore String -> Transform LCore String
-lhsT t = Transform $ method "lhs" [toJSON t]
-
--- | Apply a rewrite to the LHS of a quantified clause.
-lhsR :: Rewrite LCore -> Rewrite LCore
-lhsR r = Transform $ method "lhs" [toJSON r]
-
--- | Apply a transformation to the RHS of a quantified clause.
-rhsT :: Transform LCore String -> Transform LCore String
-rhsT t = Transform $ method "rhs" [toJSON t]
-
--- | Apply a rewrite to the RHS of a quantified clause.
-rhsR :: Rewrite LCore -> Rewrite LCore
-rhsR r = Transform $ method "rhs" [toJSON r]
-
--- | Apply a transformation to both sides of a quantified clause.
-bothT :: Transform LCore String -> Transform LCore String
-bothT t = Transform $ method "both" [toJSON t]
-
--- | Apply a rewrite to both sides of an equality, succeeding if either succeed.
-bothR :: Rewrite LCore -> Rewrite LCore
-bothR r = Transform $ method "both" [toJSON r]
+-- |
+-- lhs :: Transform LCore String -> Transform LCore String
+--   Apply a transformation to the LHS of a quantified clause.
+-- lhs :: Rewrite LCore -> Rewrite LCore
+--   Apply a rewrite to the LHS of a quantified clause.
+lhs :: LHSArgs a b => Transform a b -> Transform a b
+lhs = lhsTransform
+
+class LHSArgs a b where
+    lhsTransform :: Transform a b -> Transform a b
+    lhsTransform t = Transform $ method "lhs" [toJSON t]
+
+instance LHSArgs LCore String
+instance LHSArgs LCore LCore
```

Comment:
Why not:

```
lhs :: Transform a b -> Transform a b
lhs t = Transform $ method "lhs" [toJSON t]
```

(That is, why all the type classery if both instances use the same default method implementation anyway?)